### PR TITLE
TAUR-1125 resymboling with xignite

### PR DESCRIPTION
**Pull request by vitaly-krugl**
This PR replaces PR #93 that it uses as the starting point. The new PR refactors the logic from PRs #78 and #93 and attempts to rectify the issues that I discovered in those PRs.

NOTE that this tool is lacking tests (TAUR-1329)

### CODE REVIEW

**Comment by vitaly-krugl** [comment]
CC @scottpurdy, @jaredweiss

**Comment by vitaly-krugl** [comment]
Please review

CC @oxtopus

@jaredweiss, I would like to ask you to review this as well. This will give you an idea of what I changed corresponding to my feedback in PRs #78 and #93.

**Comment by oxtopus** [comment]
:+1: Thanks!

File: taurus.metric_collectors/taurus/metric_collectors/twitterdirect/twitter_direct_agent.py [hunk 1]
```diff
@@ -1209,12 +1188,10 @@ def _run(self):
...
       aggSec=aggSec)
 
-    # Calculate next aggregation start time using lastEmittedAggTime as base
-    lastAggStartEpoch = date_time_utils.epochFromNaiveUTCDatetime(
-      lastEmittedAggTime) + aggSec
-
-
-    nextAggEndEpoch = lastAggStartEpoch + aggSec
+    # Calculate next aggregation end time using lastEmittedAggTime as base
+    nextAggEndEpoch = (
```
Comment (jaredweiss) [inline taurus.metric_collectors/taurus/metric_collectors/twitterdirect/twitter_direct_agent.py]: I made this edit in my previous [TAUR-1125 PR](https://github.com/numenta/numenta-apps/pull/78/files#diff-1edca8ed68722259f1e76f63254cbf0dL1185) before I knew about the issue of keeping consistent time intervals in the database between updates. I linked to my original change, and I think that what I changed it from might have actually been correct. 

Either way, I think this line could benefit from a more detailed comment explaining the math.

Comment (vitaly-krugl) [inline taurus.metric_collectors/taurus/metric_collectors/twitterdirect/twitter_direct_agent.py]: @jaredweiss, I added a couple of clarification comments in the code viaf88d17bc35bc8ff1de47d5b32a5dde331a2326fd. Thanks!

**Pull request merged**